src/main/java/cope/cosmos/client/features/modules/visual/BreadcrumbsModule.java

```diff
@@ -0,0 +1,132 @@
+    public static final Setting<Float> lifespanValue = new Setting<>("Lifespan", 10F, 100F, 1000F, 0).setDescription("The lifespan of the positions in ticks").setVisible(() -> !infinite.getValue());
+    public static final Setting<Float> lineWidth = new Setting<>("LineWidth", 0.1F, 3F, 5F, 1).setDescription("The width of the lines");
+
+    // List of positions
+    private final ArrayList<Position> positions = new ArrayList<>();
+
+    public BreadcrumbsModule() {
+        super("Breadcrumbs", Category.VISUAL, "Draws a trail behind you");
+    }
+
+    @Override
+    public void onEnable() {
+        // Clear positions
+        positions.clear();
+    }
+
+    @Override
+    public void onDisable() {
+        // Clear positions
+        positions.clear();
+    }
+
+    @Override
+    public void onTick() {
+        if (!nullCheck() || mc.player.ticksExisted <= 20) {
+            // We may have just loaded into a world, so we need to clear the positions
+            positions.clear();
+            return;
+        }
+
+        // Add the player's position
+        // We are adding the player's last position so it is just behind the player, and will not be obvious on the screen (especially when elytra flying)
+        positions.add(new Position(new Vec3d(mc.player.lastTickPosX, mc.player.lastTickPosY, mc.player.lastTickPosZ)));
+
+        // Update lifespan
+        positions.forEach(position -> {
+            position.update();
+        });
+
+        // Remove positions that are too old
+        positions.removeIf(p -> !p.isAlive() && !infinite.getValue());
+    }
+
+    @Override
+    public void onRender3D() {
+        glPushMatrix();
+
+        glDisable(GL_TEXTURE_2D);
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+        glEnable(GL_LINE_SMOOTH);
+        glEnable(GL_BLEND);
+        glDisable(GL_DEPTH_TEST);
+        glLineWidth(lineWidth.getValue());
+        mc.entityRenderer.disableLightmap();
+        glBegin(GL_LINE_STRIP);
+
+        // Set line colour
+        glColor4f(ColorUtil.getPrimaryColor().getRed() / 255F, ColorUtil.getPrimaryColor().getGreen() / 255F, ColorUtil.getPrimaryColor().getBlue() / 255F, ColorUtil.getPrimaryColor().getAlpha() / 255F);
+
+        // Render positions
+        for (Position pos : positions) {
+            glVertex3d(pos.getPosition().x - mc.getRenderManager().viewerPosX, pos.getPosition().y - mc.getRenderManager().viewerPosY, pos.getPosition().z - mc.getRenderManager().viewerPosZ);
+        }
+
+        // Reset colour
+        glColor4d(1, 1, 1, 1);
+
+        glEnd();
+        glEnable(GL_DEPTH_TEST);
+        glDisable(GL_LINE_SMOOTH);
+        glDisable(GL_BLEND);
+        glEnable(GL_TEXTURE_2D);
+        glPopMatrix();
+    }
+
+    static class Position {
+        // The position's position
+        private final Vec3d position;
+
+        // The position's lifespan
+        private long lifespan = lifespanValue.getValue().longValue();
```

Comment:
i think you could do this with a map and System.currentTimeMillis()

Reply:
can i just leave it i really cba

Reply:
yea i can try to do it

Reply:
kk